addons/point_of_sale/static/src/customer_display/customer_display.xml

```diff
@@ -10,14 +10,21 @@
                 <div class="position-absolute bottom-0 mb-4 d-none d-lg-flex align-items-center ps-3 pe-2 py-1 rounded-3 text-bg-dark small">Powered by <OdooLogo style="'width: 3rem;'" monochrome="true"/></div>
             </div>
             <div class="o_customer_display_main d-flex flex-column flex-grow-1 overflow-auto">
-                <OrderWidget t-if="!order.finalized" lines="order.lines" t-slot-scope="scope" class="'gap-0 p-0 pb-3 bg-view'" style="'scroll-snap-type: y mandatory;'">
-                    <Orderline line="scope.line" class="{
-                            'o_customer_display_orderline mx-2 bg-white fs-3 rounded-0': true,
-                        }"
-                    />
-                </OrderWidget>
-                <div t-else="" class="d-flex flex-column align-items-center justify-content-center h-100 w-100 text-muted">
-                    <h3 class="w-75 mt-2 text-center">Thank you</h3>
+                <t t-if="order.lines?.length and !order.finalized">
```

Comment:
```suggestion
```

No need for this extra t-if once you changed the returned value of `emptyCartText`. You can restore the original.